### PR TITLE
docs: fix report_to_sauce being called unconditionally

### PR DESCRIPTION
**Pull request by SIGSTACKFAULT**
## Proposed changes

`report_to_sauce` is still called unconditionally, but now it silently does nothing instead of causing an error.

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the [CONTRIBUTING](https://github.com/appium/appium/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla.js.foundation/appium/appium)
- [ ] Lint and unit tests pass locally with my changes - **How do I run them?**
- [ ] I have added tests that prove my fix is effective or that my feature works - **not sure how? do i even need them?**
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

N/A

### CODE REVIEW

**Comment by SIGSTACKFAULT** [comment]
This is what I get when I try to sign the CLA: 
image

**Comment by jlipps** [comment]
You need to rebase after updating your git user information to match what is on GitHub.

**Comment by SIGSTACKFAULT** [comment]
It's asking me for my home address. Will it be public?

**Comment by jlipps** [comment]
@SIGSTACKFAULT I don't think so. Can you confirm @brianwarner?

**Comment by SIGSTACKFAULT** [comment]
I signed the CLA. no idea why EasyCLA hasn't approved.

**Comment by SIGSTACKFAULT** [comment]
A known bug, apparently.

File: sample-code/python/test/helpers.py [hunk 1]
```diff
@@ -71,6 +72,11 @@ def __save_log_type(driver, device_logger, calling_request, type):
...
 
 def report_to_sauce(session_id):
-    print("Link to your job: https://saucelabs.com/jobs/%s" % session_id)
-    passed = str(sys.exc_info() == (None, None, None))
-    sauce.jobs.update_job(session_id, passed=passed)
+    if sauce is not None:
+        print("Link to your job: https://saucelabs.com/jobs/%s" % session_id)
+        passed = str(sys.exc_info() == (None, None, None))
+        sauce.jobs.update_job(session_id, passed=passed)
+    else:
```
Comment (mykola-mokhnach) [inline sample-code/python/test/helpers.py:79]: do we need an else block that does nothing?

File: sample-code/python/test/helpers.py [hunk 2]
```diff
@@ -22,6 +22,7 @@
     # 'showIOSLog': False,
 }
 
+sauce = None
```
[the remark this answers is not in the document]
Comment (SIGSTACKFAULT) [reply sample-code/python/test/helpers.py:25]: It gets set on line 41. this is just to ensure it exists regardless of whether sauce is enabled or not.

File: sample-code/python/test/helpers.py (hunk 1 above)
Comment (SIGSTACKFAULT) [reply sample-code/python/test/helpers.py:79]: No, but putting the comment in the else block made sense at the time.

**Review by mykola-mokhnach — APPROVED** [review]

**Pull request merged**